### PR TITLE
Add intel me structures from CSS

### PR DESCRIPTION
**Pull request by ChriMarMe**
This PR shall be used to move all intel firmware related structures from 9elements/converged-security-suites to fiano.

@xaionaro @rihter007 will help as well :)

### CODE REVIEW

**Comment by ChriMarMe** [comment]
Decided to move package by package, PR by PR.

**Review by orangecms — APPROVED** [review]
just squash on merge, please, otherwise all good from my side :)

**Review by orangecms — DISMISSED** [review]

**Comment by GanShun** [comment]
Can we add this parsing to be called from NewMERegion in pkg/uefi/meregion.go? 

Side note we should probably move some of these region descriptors out of pkg/uefi into pkg/intel

**Review by GanShun — CHANGES_REQUESTED** [review]
Can we integrate this with pkg/uefi/meregion.go? I just want the parsing to happen along with everything else.

**Comment by GanShun** [comment]
If the intention is to bring everything over first then integrate after, I'm also ok with that, just let me know if that's what you all wanna do.

**Comment by orangecms** [comment]
I'd suggest that so it becomes easier to see the best way to wire things up.
After all, we're building up giant data structures, I think.

**Review by GanShun — APPROVED** [review]
Approving so we get everything in before integrating

File: pkg/intel/me/me.go
Comment (rjoleary) [inline pkg/intel/me/me.go:99]: rm this space

File: pkg/intel/me/me.go
Comment (rjoleary) [inline pkg/intel/me/me.go:149]: rm this space

File: pkg/intel/me/me.go
Comment (rjoleary) [inline pkg/intel/me/me.go:169]: Just do a `return nil, fmt.Errorf("no FlashPartitionTable found")` here and elsewhere. See https://tour.golang.org/basics/7 "Naked return statements should be used only in short functions, as with the example shown here. They can harm readability in longer functions."

File: pkg/intel/me/me.go
Comment (rjoleary) [inline pkg/intel/me/me.go:88]: You can create a new type to make this easier to deal with:

```
// String wraps around byte array to give us more control over how strings are
// serialized.
type String struct {
	Value [4]uint8
}

func (s *String) String() string {
	return strings.TrimRight(string(s.Value[:]), "\x00")
}
```

For example, see https://github.com/linuxboot/fiano/blob/master/pkg/fmap/fmap.go#L33

File: pkg/intel/me/me.go
Comment (rjoleary) [inline pkg/intel/me/me.go:148]: Can a unit test be added for this function?

**Comment by ChriMarMe** [comment]
When I get access to images this week I'll write tests for this. Getting it nice and clean before merging :)

File: pkg/intel/me/me.go
Comment (ChriMarMe) [reply pkg/intel/me/me.go:148]: Yes it could but I rather reimplement the parsing and remove this function now than implementing a test now

**Comment by ChriMarMe** [comment]
> Yes it could but I rather reimplement the parsing and remove this function now than implementing a test now

:arrow_up: 

Had a change of mind here.

**Comment by ChriMarMe** [comment]
I removed the part which searched for the structure by iterating through a hole image. This should not be the job of a parser.

**Review by orangecms — APPROVED** [review]

**Review by rjoleary — APPROVED** [review]

**Comment by ChriMarMe** [comment]
I removed the part which searched for the structure by iterating through a hole image. This should not be the job of a parser.

**Pull request merged**
